src/Microsoft.AspNetCore.Mvc.Abstractions/ModelBinding/IModelBindingContext.cs

```diff
@@ -0,0 +1,149 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.AspNetCore.Mvc.ModelBinding.Validation;
+
+namespace Microsoft.AspNetCore.Mvc.ModelBinding
+{
+    /// <summary>
+    /// A context that contains operating information for model binding and validation.
+    /// </summary>
+    public interface IModelBindingContext
```

Comment:
Should this just be an abstract class rather than an interface? Most of these properties are just get/set defaults, and the disposable could be a nested struct. 

We don't have any context interfaces that I can think of and there seem to be some good reasons for this to be a class.